pallets/eth-bridge/src/offence.rs

```diff
@@ -0,0 +1,95 @@
+use codec::{Decode, Encode};
+use frame_support::log;
+use pallet_session::{historical::IdentificationTuple, Config as SessionConfig};
+use sp_runtime::{scale_info::TypeInfo, traits::Convert, Perbill};
+use sp_staking::{
+    offence::{Kind, Offence, ReportOffence},
+    SessionIndex,
+};
+use sp_std::prelude::*;
+
+use crate::Event;
+
+#[derive(PartialEq, Eq, Clone, Debug, Encode, Decode, TypeInfo)]
+pub enum CorroborationOffenceType {
+    ChallengeAttemptedOnSuccessfulTransaction,
+    ChallengeAttemptedOnUnsuccessfulTransaction,
+}
+
+#[derive(PartialEq, Clone, Debug, Encode, Decode, TypeInfo)]
+pub struct CorroborationOffence<Offender> {
+    pub session_index: SessionIndex,
+    pub offenders: Vec<Offender>,
+    pub offence_type: CorroborationOffenceType,
+    pub validator_set_count: u32,
+}
+
+impl<Offender: Clone> Offence<Offender> for CorroborationOffence<Offender> {
+    const ID: Kind = *b"bridge:::offence";
+    type TimeSlot = SessionIndex;
+
+    fn offenders(&self) -> Vec<Offender> {
+        self.offenders.clone()
+    }
+
+    fn session_index(&self) -> SessionIndex {
+        self.session_index
+    }
+
+    fn validator_set_count(&self) -> u32 {
```

Comment:
Why do we need this? Isn't the result of this function to enable us to call `obj.validator_set_count` from the outside? But if that is the same name  of the internal field, why is this not redundant?

Reply:
this is the implementation of the `Offence` trait, so this implementation is just a stub, and yes it is redundant